oximeter/collector/src/lib.rs

```diff
@@ -157,6 +159,49 @@ pub struct OximeterArguments {
     pub address: SocketAddrV6,
 }
 
+// Provides an alternative to the DNS resolver for cases where we want to
+// contact a backend without performing resolution.
+struct SingleHostResolver {
```

Comment:
This exists so the old "connect to single address" stuff still works.

It is an alternative to DNS resolution, and just connects to a single address as a backend.